### DISCUSSION TITLE
How can we use YAGSL with MAXSwerve encoders that go through the SparkMAX motor controller?

### DISCUSSION THREAD

**Discussion by isubsmith** [BroncBotz3481/YAGSL-Example#58] [Q&A]
When we try using YAGSL with the encoders set to "throughbore" and the IDs set to the CANIDs of the motor controllers, the driving and turning motors spin very rapidly. Doing it with the PID values of the driving and turning motors set to 0 makes the driving motors spin very rapidly and the turning motors not spin at all. Is it possible to use YAGSL with MAXSwerve?

**Comment by thenetworkgrinch** [comment]
It should be, the encoder pulse per revolution with throughbores attached to sparkmax's should be 1btw

**Comment by mikejdavey1** [comment]
Did you get this working?  We are trying this out on the offseason and just powered our machine up..  
Similar issue, on startup 3 of our 4 modules turn in about 45 degree increments, then the slightest bump and 3 of the 4 drive motors take off full speed.. I have to figure out what is happending on the 4th module as that doesn't do anything.

Using the example code with just the neo configurations changed for our IDs and throughbore - the other issue is that the Throughbore configuration actually takes analog input channels up even though it doesn't use them.  the IDs you put in are assigned as analog inputs, as we first had them defaults (11, 12 etc) and crashed the code, then had them all set to 0 zeros and got analog channel already assigned errors.

Can someone share their module configures for MAXSwerve?  We are also using a NavX, we are just testing on blocks.

**Reply by armans-code** [reply]
Did you find a fix? We are also running into the "analog channel already assigned" errors. We have through-bore encoders connected to the spark max. Not sure what to put for the Absolute Encoder ID or CAN Bus values within the JSON config. We tried both setting it to 0 and setting it to the same value as its corresponding sparkmax controller.

**Reply by thenetworkgrinch** [reply]
Its supposed to be "attached" the CANBus and CAN ID values are ignored when you do this, fortunately there has been a team which has a PR open for it. I havent merged it yet.

https://github.com/BroncBotz3481/YAGSL-Example/pull/77
